packages/component-base/src/active-mixin.js

```diff
@@ -27,19 +27,15 @@ const ActiveMixinImplementation = (superclass) =>
     ready() {
       super.ready();
 
-      this._addEventListenerToNode(this, 'down', () => {
-        if (this.disabled) return;
-
-        this._setActive(true);
+      this._addEventListenerToNode(this, 'down', (event) => {
+        if (this._shouldSetActive(event)) {
+          this._setActive(true);
+        }
       });
 
       this._addEventListenerToNode(this, 'up', () => {
         this._setActive(false);
       });
-
-      this.addEventListener('blur', () => {
-        this._setActive(false);
-      });
```

Comment:
Having this `blur` listener causes the checkbox to lose the `active` attribute in case the user switches the checkbox by clicking on the label (for some reason, there is fired the blur event on `<vaadin-checkbox>` (!!) while moving focus between the input and the label).

Didn't have enough time to investigate that all the way but I assume the `blur` listener can be removed altoghether as it originally came to the mixin from the button component. However, even for the button, it doesn't look relevant since we changed its implementation so that there are not any native buttons.